src/Sarif.Multitool/Rules/SARIF1022.PreferMessageIdInsteadOfMessageText.cs

```diff
@@ -0,0 +1,108 @@
+
+        public override FailureLevel DefaultLevel => FailureLevel.Error;
+
+        /// <summary>
+        /// SARIF1019
+        /// </summary>
+        public override string Id => RuleId.PreferMessageIdInsteadOfMessageText;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF1022_RuleMetadataDoesntIncludeMessageId),
+            nameof(RuleResources.SARIF1022_RuleMetadataIsNotConsistentWithArguments),
+            nameof(RuleResources.SARIF1022_ResultShouldUseMessageIdInsteadOfMessageText)
+        };
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            this.currentRules = run?.Tool?.Driver?.Rules;
+        }
+
+        protected override void Analyze(Result result, string resultPointer)
+        {
+            // if message.text is not null, we have a problem
+            if (!string.IsNullOrEmpty(result.Message.Text))
+            {
+                LogResult(resultPointer, nameof(RuleResources.SARIF1022_ResultShouldUseMessageIdInsteadOfMessageText), result.Message.Text);
+                return;
+            }
+
+            // if message.id is not null, we have to check if that exists in rules
+            if (!string.IsNullOrEmpty(result.Message.Id))
+            {
+                // we dont have rules or we didnt find a match, we have a problem
+                if (this.currentRules == null || this.currentRules.Count == 0 || this.currentRules.All(r => !r.MessageStrings.ContainsKey(result.Message.Id)))
+                {
+                    LogResult(resultPointer, nameof(RuleResources.SARIF1022_RuleMetadataDoesntIncludeMessageId), result.Message.Id);
+                    return;
+                }
+
+                // we have rules, we find it but, the count of parameters are invalid
+                string messageText = this.currentRules
+                    .First(r => r.MessageStrings.ContainsKey(result.Message.Id))
+                    .MessageStrings[result.Message.Id].Text;
+                if (!IsPlacehodlerInTextValid(messageText, result.Message.Arguments?.Count ?? 0))
+                {
+                    LogResult(resultPointer, nameof(RuleResources.SARIF1022_RuleMetadataIsNotConsistentWithArguments), messageText, result.Message.Arguments.Count.ToString());
+                    return;
+                }
+            }
+        }
+
+        private bool IsPlacehodlerInTextValid(string text, int arguments)
+        {
+            for (int i = 0; i < arguments - 1; i++)
+            {
+                if (!text.Contains($"{i}"))
+                {
+                    return false;
+                }
+            }
+
+            // checking the quantity of matches with quantity of arguments
+            MatchCollection matchCollection = Regex.Matches(text, "{\\d+}");
+            if (matchCollection.Count != arguments)
```

Comment:
>arguments [](start = 41, length = 9)

BUG: It's possible for a message to mention the same placeholder more than once. For an example, see my suggested message:

"{0}: The result message contains {1} message arguments for the message string with id '{2}', but the metadata for rule '{3}' defines message string '{2}' to take {4} arguments."

You see that placeholder {2} (the message string id) occurs twice in the message string.